code/drasil-code/Language/Drasil/Code/Imperative/Build/Import.hs

```diff
@@ -3,63 +3,61 @@ module Language.Drasil.Code.Imperative.Build.Import (
 ) where
 
 import Language.Drasil.Code.Code (Code(..))
-import Language.Drasil.Code.Imperative.AST (Label, Module(Mod), notMainModule,
-  Package(Pack))
+import Language.Drasil.Code.Imperative.New (Label)
+import Language.Drasil.Code.Imperative.Helpers (tripThird)
 import Language.Drasil.Code.Imperative.Build.AST (BuildConfig(BuildConfig),
   BuildDependencies(..), Ext(..), includeExt, NameOpts, nameOpts, packSep,
   Runnable(Runnable), BuildName(..), RunType(..))
-import Language.Drasil.Code.Imperative.LanguageRenderer (Config, buildConfig,
-  ext, runnable)
 
 import Build.Drasil ((+:+), genMake, makeS, MakeString, mkFile, mkRule,
   mkCheckedCommand, mkFreeVar, RuleTransformer(makeRule))
 
+import Text.PrettyPrint.HughesPJ (Doc)
 import Data.List.Utils (endswith)
 import Data.Maybe (maybe, maybeToList)
 
-data CodeHarness = Ch Config Package Code
+data CodeHarness = Ch (Maybe BuildConfig) Runnable [String] ([(Doc, Label, Bool)], Label) Code
 
 instance RuleTransformer CodeHarness where
-  makeRule (Ch c m co@(Code code)) = [
-    mkRule buildTarget (map (const $ renderBuildName c m nameOpts nm) $ maybeToList $
-      buildConfig c) []
+  makeRule (Ch b r e m co@(Code code)) = [
+    mkRule buildTarget (map (const $ renderBuildName e m nameOpts nm) $ maybeToList b) []
     ] ++
     maybe [] (\(BuildConfig comp bt) -> [
-    mkFile (renderBuildName c m nameOpts nm) (map (makeS . fst) code) [
-      mkCheckedCommand $ foldr (+:+) mempty $ comp (getCompilerInput bt c m co) $
-        renderBuildName c m nameOpts nm
+    mkFile (renderBuildName e m nameOpts nm) (map (makeS . fst) code) [
+      mkCheckedCommand $ foldr (+:+) mempty $ comp (getCompilerInput bt e m co) $
+        renderBuildName e m nameOpts nm
       ]
-    ]) (buildConfig c) ++ [
+    ]) b ++ [
     mkRule (makeS "run") [buildTarget] [
-      mkCheckedCommand $ buildRunTarget (renderBuildName c m no nm) ty +:+ mkFreeVar "RUNARGS"
+      mkCheckedCommand $ buildRunTarget (renderBuildName e m no nm) ty +:+ mkFreeVar "RUNARGS"
       ]
     ] where
-      (Runnable nm no ty) = runnable c
+      (Runnable nm no ty) = r
       buildTarget = makeS "build"
 
-renderBuildName :: Config -> Package -> NameOpts -> BuildName -> MakeString
-renderBuildName _ (Pack _ m) _ BMain = makeS $ getMainModule m
-renderBuildName _ (Pack l _) _ BPackName = makeS l
-renderBuildName c p o (BPack a) = renderBuildName c p o BPackName <> makeS (packSep o) <> renderBuildName c p o a
-renderBuildName c p o (BWithExt a e) = renderBuildName c p o a <> if includeExt o then renderExt c e else makeS ""
+renderBuildName :: [String] -> ([(Doc, Label, Bool)], Label) -> NameOpts -> BuildName -> MakeString
```

Comment:
I'm uncomfortable with hard-coding `(Doc, Label, Bool)` here. It's likely to change. Should be abstracted.

Reply:
Replaced (Doc, Label, Bool) with a new record type in 77139c28eaa4cc4d2b155e47de2fcfe1c7682fbb.